### PR TITLE
Refactor filters and enable more unit tests

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR:
* Refactors filters to allow internal filters to be public filters that can provide transforms or VRL
* Fixes the output factory test
* Add filters: parsejson, multi-line exception, openshift labels

is blocked by #2499 

### Links
* https://issues.redhat.com/browse/LOG-5367
* https://issues.redhat.com/browse/LOG-5553
* https://issues.redhat.com/browse/LOG-5554
* https://issues.redhat.com/browse/LOG-5555

cc @cahartma @vparfonov @Clee2691

### CODE REVIEW

File: api/observability/v1/filter_types.go [hunk 1]
```diff
@@ -16,12 +16,12 @@ package v1
...
 //
-// +kubebuilder:validation:Enum:=addLabels;detectMultiline;drop;kubeAPIAudit;parse;prune
+// +kubebuilder:validation:Enum:=openShiftLabels;detectMultilineException;drop;kubeAPIAudit;parse;prune
 type FilterType string
 
 // Filter type constants, must match JSON tags of FilterTypeSpec fields.
 const (
-	FilterTypeAddLabels       FilterType = "addLabels"
+	FilterTypeOpenshiftLabels FilterType = "openShiftLabels"
 	FilterTypeDetectMultiline FilterType = "detectMultiline"
```
Comment (vparfonov) [inline api/observability/v1/filter_types.go:25]: Should here be detectMultiline**Exception** ?

File: internal/generator/vector/filter/openshift/viaq/filter.go [hunk 2]
```diff
@@ -51,72 +45,79 @@ if .log_source == "%s" {
...
 		vrls = append(vrls, journalLogs())
 	}
 	return vrls
 }
 
-func HasJournalSource(inputs []logging.InputSpec) bool {
-	return hasSource(inputs, logging.InputNameInfrastructure, logging.InfrastructureSourceNode)
+func HasJournalSource(inputs []obs.InputSpec) bool {
+	for _, i := range inputs {
+		if i.Type == obs.InputTypeInfrastructure && i.Infrastructure != nil && (set.New(i.Infrastructure.Sources...).Has(obs.InfrastructureSourceNode) || len(i.Infrastructure.Sources) == 0) {
```
Comment (vparfonov) [inline internal/generator/vector/filter/openshift/viaq/filter.go:97]: nit:
```suggestion
		if i.Type == obs.InputTypeInfrastructure && i.Infrastructure != nil && (len(i.Infrastructure.Sources) == 0 || set.New(i.Infrastructure.Sources...).Has(obs.InfrastructureSourceNode)) {
```

File: api/observability/v1/filter_types.go (hunk 1 above)
Comment (jcantrill) [reply api/observability/v1/filter_types.go:25]: Agree.  I have this as part of a commit that has not been pushed yet

Comment (jcantrill) [reply api/observability/v1/filter_types.go:25]: Commit was on this branch.. updated here

**Review by cahartma — COMMENTED** [review]
/lgtm

**Pull request merged**